[matter] Exception when using the test button of my Heiman smoke detector

@digitaldan : I succeeded adding a first Thread device to openHAB, a smoke detector from Heiman.
After discovering the device and linking all channels, everything looks good, even if all these provided channels are relatively confusing. Maybe some should be advanced channels.

As I can't do a lot of things, I just tried to push the test button on the device.
Here are the exceptions I got:
```
2026-01-11 11:49:59.798 [WARN ] [mmon.WrappedScheduledExecutorService] - Scheduled runnable ended with an exception:
com.google.gson.JsonSyntaxException: java.lang.IllegalStateException: Expected BEGIN_OBJECT but was STRING at path $
        at com.google.gson.internal.bind.ReflectiveTypeAdapterFactory$Adapter.read(ReflectiveTypeAdapterFactory.java:523) ~[?:?]
        at com.google.gson.Gson.fromJson(Gson.java:1359) ~[?:?]
        at com.google.gson.Gson.fromJson(Gson.java:1472) ~[?:?]
        at com.google.gson.Gson.fromJson(Gson.java:1443) ~[?:?]
        at com.google.gson.internal.bind.TreeTypeAdapter$GsonContextImpl.deserialize(TreeTypeAdapter.java:200) ~[?:?]
        at org.openhab.binding.matter.internal.client.MatterWebsocketClient$EventTriggeredMessageDeserializer.deserialize(MatterWebsocketClient.java:678) ~[?:?]
        at org.openhab.binding.matter.internal.client.MatterWebsocketClient$EventTriggeredMessageDeserializer.deserialize(MatterWebsocketClient.java:1) ~[?:?]
        at com.google.gson.internal.bind.TreeTypeAdapter.read(TreeTypeAdapter.java:95) ~[?:?]
        at com.google.gson.Gson.fromJson(Gson.java:1359) ~[?:?]
        at com.google.gson.Gson.fromJson(Gson.java:1472) ~[?:?]
        at com.google.gson.Gson.fromJson(Gson.java:1416) ~[?:?]
        at org.openhab.binding.matter.internal.client.MatterWebsocketClient.lambda$2(MatterWebsocketClient.java:298) ~[?:?]
        at java.util.concurrent.Executors$RunnableAdapter.call(Unknown Source) ~[?:?]
        at java.util.concurrent.FutureTask.run(Unknown Source) ~[?:?]
        at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(Unknown Source) ~[?:?]
        at java.util.concurrent.ThreadPoolExecutor.runWorker(Unknown Source) ~[?:?]
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(Unknown Source) ~[?:?]
        at java.lang.Thread.run(Unknown Source) [?:?]
Caused by: java.lang.IllegalStateException: Expected BEGIN_OBJECT but was STRING at path $
        at com.google.gson.internal.bind.JsonTreeReader.expect(JsonTreeReader.java:184) ~[?:?]
        at com.google.gson.internal.bind.JsonTreeReader.beginObject(JsonTreeReader.java:101) ~[?:?]
        at com.google.gson.internal.bind.ReflectiveTypeAdapterFactory$Adapter.read(ReflectiveTypeAdapterFactory.java:512) ~[?:?]
        ... 17 more
2026-01-11 11:49:59.798 [WARN ] [mmon.WrappedScheduledExecutorService] - Scheduled runnable ended with an exception:
com.google.gson.JsonSyntaxException: java.lang.IllegalStateException: Expected BEGIN_OBJECT but was STRING at path $
        at com.google.gson.internal.bind.ReflectiveTypeAdapterFactory$Adapter.read(ReflectiveTypeAdapterFactory.java:523) ~[?:?]
        at com.google.gson.Gson.fromJson(Gson.java:1359) ~[?:?]
        at com.google.gson.Gson.fromJson(Gson.java:1472) ~[?:?]
        at com.google.gson.Gson.fromJson(Gson.java:1443) ~[?:?]
        at com.google.gson.internal.bind.TreeTypeAdapter$GsonContextImpl.deserialize(TreeTypeAdapter.java:200) ~[?:?]
        at org.openhab.binding.matter.internal.client.MatterWebsocketClient$EventTriggeredMessageDeserializer.deserialize(MatterWebsocketClient.java:678) ~[?:?]
        at org.openhab.binding.matter.internal.client.MatterWebsocketClient$EventTriggeredMessageDeserializer.deserialize(MatterWebsocketClient.java:1) ~[?:?]
        at com.google.gson.internal.bind.TreeTypeAdapter.read(TreeTypeAdapter.java:95) ~[?:?]
        at com.google.gson.Gson.fromJson(Gson.java:1359) ~[?:?]
        at com.google.gson.Gson.fromJson(Gson.java:1472) ~[?:?]
        at com.google.gson.Gson.fromJson(Gson.java:1416) ~[?:?]
        at org.openhab.binding.matter.internal.client.MatterWebsocketClient.lambda$2(MatterWebsocketClient.java:298) ~[?:?]
        at java.util.concurrent.Executors$RunnableAdapter.call(Unknown Source) ~[?:?]
        at java.util.concurrent.FutureTask.run(Unknown Source) ~[?:?]
        at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(Unknown Source) ~[?:?]
        at java.util.concurrent.ThreadPoolExecutor.runWorker(Unknown Source) ~[?:?]
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(Unknown Source) ~[?:?]
        at java.lang.Thread.run(Unknown Source) [?:?]
Caused by: java.lang.IllegalStateException: Expected BEGIN_OBJECT but was STRING at path $
        at com.google.gson.internal.bind.JsonTreeReader.expect(JsonTreeReader.java:184) ~[?:?]
        at com.google.gson.internal.bind.JsonTreeReader.beginObject(JsonTreeReader.java:101) ~[?:?]
        at com.google.gson.internal.bind.ReflectiveTypeAdapterFactory$Adapter.read(ReflectiveTypeAdapterFactory.java:512) ~[?:?]
        ... 17 more
```
Looks like a string is returned when you expect a JSON value.
Here are how the items were changed during the test:
```
2026-01-11 11:49:55.706 [INFO ] [openhab.event.ItemStateChangedEvent ] - Item 'HEIMAN_Smoke_sensor_RDC_Power_Source_Expressed_State' changed from 0 to 4 (source: org.openhab.core.thing$matter:node:openhab:254940193XXXXXXXXXX:1#smokecoalarm-expressedstate)
2026-01-11 11:49:55.708 [INFO ] [openhab.event.ItemStateChangedEvent ] - Item 'HEIMAN_Smoke_sensor_RDC_Power_Source_Test_In_Progress' changed from OFF to ON (source: org.openhab.core.thing$matter:node:openhab:254940193XXXXX:1#smokecoalarm-testinprogress)
2026-01-11 11:49:59.813 [INFO ] [openhab.event.ItemStateChangedEvent ] - Item 'HEIMAN_Smoke_sensor_RDC_Power_Source_Expressed_State' changed from 4 to 0 (source: org.openhab.core.thing$matter:node:openhab:254940193XXXXX:1#smokecoalarm-expressedstate)
2026-01-11 11:49:59.814 [INFO ] [openhab.event.ItemStateChangedEvent ] - Item 'HEIMAN_Smoke_sensor_RDC_Power_Source_Test_In_Progress' changed from ON to OFF (source: org.openhab.core.thing$matter:node:openhab:254940193XXXXX:1#smokecoalarm-testinprogress)
```

--- a/bundles/org.openhab.binding.matter/matter-server/src/util/Json.ts
+++ b/bundles/org.openhab.binding.matter/matter-server/src/util/Json.ts
@@ -57,7 +57,7 @@ export function toJSON(data: any) {
             return Bytes.toHex(value);
         }
         if (value === undefined) {
-            return "undefined";
+            return null;
         }
         return value;
     });

